### PR TITLE
Add run_constrained

### PR DESCRIPTION
**Pull request by dcherian**
From https://github.com/pydata/xarray/blob/main/ci/requirements/min-all-deps.yml with only netcdf4, h5netcdf, zarr backends specified.

Closes https://github.com/pydata/xarray/issues/7467

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)

### CODE REVIEW

**Comment by dcherian** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -25,6 +25,28 @@ requirements:
...
+    - h5netcdf >=0.13
+    - h5py >=3.6
+    - hdf5 >=1.12
+    - iris >=3.1
+    - matplotlib-base >=3.5
+    - nc-time-axis >=1.4
+    - netcdf4 >=1.5.7
+    - numba >=0.55
+    - pint >=0.18
+    - rasterio >=1.2
```
Comment (dcherian) [inline recipe/meta.yaml:44]:
```suggestion
```

didn't intend to do this.

**Comment by jhamman** [comment]
To check my understanding of what this does, this will warn (or error?) when a user tries to install xarray into a environment where one of the listed optional dependencies is out of range from what is listed in `run_constrained`?

**Comment by ocefpaf** [comment]
@jhamman that will error out with a package conflict message. It is, in a way, the best way to control optional dependencies version.

**Comment by ocefpaf** [comment]
> And when creating a new env, it'll ensure these minimum versions even if the env file doesn't list them?

Nope. The user must list the optional dependencies. The user can drop the min versions pin though, the solver will do te right thing and respect the run constraints in xarray.

**Comment by dcherian** [comment]
Right that's what I meant. Sorry for nbeing unclear.

The user lists all the packages they want; and the solver will respect xarray's run constraints. Sounds like a feature!

We should have a policy on what to list here though: https://github.com/pydata/xarray/issues/7573

Should Xarray limit iris for example? It seems like almost no one is converting xarray datasets to iris or vice versa.

**Comment by ocefpaf** [comment]
> The user lists all the packages they want; and the solver will respect xarray's run constraints. Sounds like a feature!

Yeah. We, conda-forge, should probably add those automatically for python projects in our recipes via grayskull. The problem is that most Python packages use this so loosely that we will probably create packages that are too "rigid" to install rather than more stable solves.

TL;DR the xarray metadata is solid and that makes sense here. Doing that for all packages, not so much.

**Comment by hmaarrfk** [comment]
is pandas 2 safe?

**Comment by ocefpaf** [comment]
> is pandas 2 safe?

For the xarray version in this PR? No. But if we are to merge this we would rebase and the latest version of xarray already pins pandas accordingly.

**Comment by jhamman** [comment]
closed by #93
